### ISSUE TITLE
Running multiple pingers in concurrent causes true ping on wrong ip

### ISSUE THREAD

**Issue by enzofoucaud** [go-ping/ping#129]
Hello,

I'm using this package for a project and I have a problem when I execute multiple pings.
I have 11 ping in concurrent and I only have three reachable IPs out of the eleven.

When i execute my go routine i have some ip out which have a true state.
The pings are random over time.
I tested with a bash ping exec directly and I don't have this problem.

Why ?

OS Running : Linux Ubuntu 20.04 and 18.04 / Windows 10
Go : 1.15.6 and 1.15.17

Code to catch true ping :
`
if state.state == true && target != "8.8.8.8" && target != "10.0.202.50" && target != "10.0.202.51" {
 log.Error(target, " : ", state.state)
}
`
Log : 
> ERRO[0028] 10.0.202.57 : true
> ERRO[0046] 10.0.202.52 : true
> ERRO[0088] 10.0.202.54 : true
> ERRO[0096] 10.0.202.58 : true
> ERRO[0100] 10.0.202.52 : true

@CHTJonas

**Comment by enzofoucaud** [comment]
@CHTJonas

**Comment by CHTJonas** [comment]
Hello @enzofoucaud! Thanks for the little code snippet -- would it be possible to provide an entire reproducible case? If you could upload a `.go` source file or paste the code into `<details>...</details>` tags in the comments section that would be helpful, thanks!

**Comment by enzofoucaud** [comment]
Hello @CHTJonas ,

This morning I tried seveI created a snippet coderal things that are interesting to note!
I created a snippet code to show the error it made.
I have two different codes. My main project and test_ping

On WSL2 main project : I have true pings when they must be false AND when I launch a heavy task such as a VM startup, the pings are more substantial.
> ERRO[0028] 10.0.202.57 : true
> ERRO[0046] 10.0.202.52 : true
> ERRO[0088] 10.0.202.54 : true
> ERRO[0096] 10.0.202.58 : true
> ERRO[0100] 10.0.202.52 : true

On WSL2 test_ping : All pings are false, even those that should be true

Windows 10 v20H2 : Main project and test_ping run perfectly without problems.

Ubuntu server 20.04.1 and desktop 20.04.1 Main project and test_ping run perfectly without problems

[test_ping.zip](https://github.com/go-ping/ping/files/5893045/test_ping.zip)

**Comment by CHTJonas** [comment]
Thanks for that ZIP archive! I have taken your code, simplified it slightly and tested but I'm unable to replicate. Can I just confirm, with `pinger.SetPrivileged(true)` you need to either run as root or `sudo setcap cap_net_raw=+ep /path/to/your/compiled/binary` otherwise it won't work and you'll observe mystic packet loss.

I've pasted my `main.go` test code in the section below and am executing via `go build main.go && sudo ./main` on the command line.

<details>

```go
package main

import (
	"fmt"
	"log"
	"time"

	"github.com/go-ping/ping"
)

type Service struct {
	IP     string
	Status bool
}

type equipement struct {
	nameID string
	state  bool
}

var servicesList = map[string]*Service{
	"server0":  &Service{IP: "8.8.8.8"},
	"server1":  &Service{IP: "1.10.16.1"},
	"server2":  &Service{IP: "1.10.16.2"},
	"server3":  &Service{IP: "1.10.16.3"},
	"server4":  &Service{IP: "1.10.16.4"},
	"server5":  &Service{IP: "1.10.16.5"},
	"server6":  &Service{IP: "1.10.16.6"},
	"server7":  &Service{IP: "1.10.16.7"},
	"server8":  &Service{IP: "1.10.16.8"},
	"server9":  &Service{IP: "1.10.16.9"},
	"server10": &Service{IP: "1.1.1.1"},
}

func main() {
	ticker := time.NewTicker(500 * time.Millisecond)
	sender := make(chan equipement)
	go modifyStatus(sender)
	for {
		<-ticker.C
		for name, serv := range servicesList {
			fmt.Println("Iterating over", name, "with IP address", serv.IP)
			doPing(serv.IP, name, sender)
		}
	}
}

func modifyStatus(sender chan equipement) {
	for {
		state := <-sender
		servicesList[state.nameID].Status = state.state
	}
}

func doPing(target string, name string, sender chan equipement) {
	state := equipement{
		nameID: name,
		state:  false,
	}
	pinger, err := ping.NewPinger(target)
	if err != nil {
		log.Fatal(err)
	}
	pinger.SetPrivileged(true)
	pinger.Timeout = time.Duration(400 * time.Millisecond)
	pinger.Count = 1
	fmt.Println("Pinging", name, "with IP address", target)
	pinger.Run()
	stats := pinger.Statistics()
	state.state = stats.PacketsRecv > 0 && stats.PacketsSent > 0 && stats.PacketLoss == 0
	fmt.Println("Pinged", name, "with IP address", target, "and got these stats:", stats.PacketsSent, stats.PacketsRecv)
	if state.state == true && target != "8.8.8.8" && target != "1.1.1.1" {
		log.Fatal(target, " : ", state.state)
	}
	fmt.Println(state)
	sender <- state
}
```

</details>

**Comment by bestoml** [comment]
Maybe it has been fixed yesterday.
-	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	r := rand.New(rand.NewSource(getSeed()))

The old source will cause all the pinger has the same id.

**Comment by enzofoucaud** [comment]
@CHTJonas where you test the code ?

I can't test now my main project and test code for the moment because vacancy !
Next week :)

@bestoml when i'm comeback i try it

**Comment by enzofoucaud** [comment]
@CHTJonas Hello i'm back,

I test your code and i don't have any error too .. 
This is a piece of my main code and I set the privileges to true
`
state := equipement{
  nameID: name,
  state:  false,
  }
  pinger, err := ping.NewPinger(target)
  pinger.SetPrivileged(true)
  if err != nil {
  log.Error(err)
  }
  pinger.Timeout = time.Duration(400 * time.Millisecond)
  pinger.Count = 1
  pinger.Run()
  stats := pinger.Statistics()
`

When I run my project, I still get random ping errors. And for no IP can be reached. Only with WSL2.
I tried on other machines in the company and the problem is the same.
> ERRO[0007] 10.0.202.52 : true
> ERRO[0037] 10.0.202.56 : true
